src/protocol.rs

```diff
@@ -190,6 +190,62 @@ pub fn read_info_field(device: &HidDevice, field: InfoField) -> Result<String, E
         .to_owned())
 }
 
+pub fn run_tap_commands(device: &HidDevice) -> Result<(), Error> {
+    const TAP_REPORT_ID: u8 = 2;
+    const TAP_REPORT_LEN: usize = 2048;
+
+    loop {
+        print!("> ");
+        std::io::stdout().flush().unwrap();
+        let mut tap_command = String::new();
+        std::io::stdin()
+            .read_line(&mut tap_command)
+            .expect("Failed to read line");
+
+        tap_command = tap_command.trim().to_string();
```

Comment:
You don't need `to_string()` here. Just use a new `&str` variable instead of reassigning `tap_command`.